### ISSUE TITLE
[BUG] Bluetooth Play/Pause issue

### ISSUE THREAD

**Issue by uploadsjuicers** [LeoKlaus/plappa#195]
**Describe the bug**

Over Bluetooth, pressing the play/pause button on my speaker only starts playback; it doesn't pause when I press it again. This issue occurs with both a JBL wireless speaker and the touchscreen controls in my vehicle. In the vehicle, the play icon never changes to pause during playback. However, tapping play does resume paused tracks.

In essence, the Bluetooth control for both devices starts playback but lacks the expected pausing functionality when pressed again.

**To Reproduce**
Connect to Bluetooth, start audio. Press play button again (to pause)
Audio doesn't pause

**Expected behavior**
When using other apps on the same device, the button both plays and pauses audio. With Plappa I only have play functionality. 

**Environment(please complete the following information):**
 - Device: iPhone 15 pro
 - OS:iOS 18
 - Version: latest
 - Server Version: Audio Bookshelf latest

**Comment by LeoKlaus** [comment]
That is curious. First off, can you make sure that you are actually on the latest version of plappa?
You can see the version number in the "About plappa" section of the settings. The latest AppStore release is 1.4.3, TestFlight is on 1.4.4.

The remote controls know three commands: play, pause and playpause, each of which are configured correctly in plappa. The behavior you describe could only occur if both your bluetooth speaker and car stereo treated the play/pause button as a play button only (my JBL Charge 3 doesn't do this).

**Comment by uploadsjuicers** [comment]
Yes, sorry I'm on the latest App Store release 1.4.3. I agree it's curious because when I use the Apple Music app I don't have this issue.

I also just tried it with another bluetooth device (headphones) and the play/pause functionality worked correctly. Does this mean it's likely an issue with the other devices?

Thanks!

**Comment by BobSwarner** [comment]
I am seeing the exact same issue.  I am using plappa version 1.4.3 on iOS 18.  Pressing the play/pause button on my headset only starts playback; it doesn't pause when I press it again.

The play/pause functionality works fine with my podcast and music apps.

Update: It does work correctly on my Bluetooth speaker, but not with my headset.  The headset is Open Run by Shokz.

**Comment by uploadsjuicers** [comment]
I've just learned something else about this issue. I have been listening at 1.2x speed. When I change the playback speed back to 1.0 the issue goes away and the play/pause functionality works as expected.

**Comment by LeoKlaus** [comment]
Didn't mean to close this yet...

Thanks for the tip with the playback speed @uploadsjuicers!

It turns out that when using a bluetooth speaker with a playback rate higher than 1.0, the system calls playCommand, not playPauseCommand or pauseCommand. For playback rates <=1.0, the system correctly calls pauseCommand.

I have no idea why this would be the case, but I have just reverted the behavior to play/pause, no matter what command the system calls. I have just released 1.4.4 build 58 to TestFlight, let me know if that worked!

**Comment by BobSwarner** [comment]
Great news, @LeoKlaus.  Please send me whatever I need to access the TestFlight version, and I will check it out.

**Comment by uploadsjuicers** [comment]
I tried this update and it does make the play/pause work correctly. I will note, though that in my car's touchscreen the icon doesn't switch after pressing the button. It only ever shows the play icon. But if I press the play icon it does stop the already playing audio. This icon swaps correctly at 1.0 speed. Functionally, it does work as expected now, so thanks for the fix! I appreciate your speedy responses.

**Comment by uploadsjuicers** [comment]
> Great news, @LeoKlaus. Please send me whatever I need to access the TestFlight version, and I will check it out.

here's the link https://testflight.apple.com/join/oSAekRD5

**Comment by LeoKlaus** [comment]
> I will note, though that in my car's touchscreen the icon doesn't switch after pressing the button. It only ever shows the play icon. But if I press the play icon it does stop the already playing audio. 

Weird, does the same happen with the controls on the lock screen of your phone? Seems like iOS doesn't recognize the player as playing with rate above 1.0 for some reason...

**Comment by uploadsjuicers** [comment]
I do not have this issue on the Lock Screen, only with my car's Bluetooth and only at speeds over 1.0. I have not checked speeds slower than 1.0. At 1.0 speed the play button turns into a pause button during playback.

**Comment by BobSwarner** [comment]
I have tested the 1.4.4 version and it does resolve this issue on my headset.   I will continue to exercise it, but looks great so far.  Thank you.

**Comment by hughrawlinson** [comment]
Confirmed fix on my headphones in 1.4.4 (Bose 700, iPhone 12 running 18.0.1)

**Comment by cwhite616** [comment]
Just upgraded to 1.5.0 through TestFlight and am having this same bug… it only happens when playback is >1, and strangely enough we can touch to play, but not touch to pause.

iOS 18.0.1, iPhone 13, some cheap Bluetooth headphones that nonetheless work with every other audio app 🙃
